src/analysis/retail/priest/discipline/modules/spells/TwilightEquilibrium.tsx

```diff
@@ -0,0 +1,135 @@
+import { encodeEventTargetString } from 'parser/shared/modules/Enemies';
+
+const TE_INCREASE = 0.15;
+
+class TwilightEquilibrium extends Analyzer {
+  healing = 0;
+  damage = 0;
+  ptwTargets: Set<string> = new Set<string>();
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS_PRIEST.TWILIGHT_EQUILIBRIUM_TALENT.id);
+
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(AtonementAnalyzer.atonementEventFilter, this.onAtone);
+    this.addEventListener(
+      Events.applydebuff.by(SELECTED_PLAYER).spell([SPELLS.PURGE_THE_WICKED_TALENT]),
+      this.onPTWApply,
+    );
+
+    this.addEventListener(
+      Events.refreshdebuff.by(SELECTED_PLAYER).spell(SPELLS.PURGE_THE_WICKED_BUFF),
+      this.onPTWApply,
+    );
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onDamage);
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.PURGE_THE_WICKED_BUFF),
+      this.onPTWDotDamage,
+    );
+  }
+
+  onAtone(event: AtonementAnalyzerEvent) {
+    const { damageEvent, healEvent } = event;
+
+    if (damageEvent?.ability.guid === SPELLS.PURGE_THE_WICKED_BUFF.id) {
+      if (this.ptwTargets.has(encodeEventTargetString(damageEvent) || '')) {
+        this.healing += calculateEffectiveHealing(healEvent, TE_INCREASE);
+      }
+    }
+
+    if (HOLY_DAMAGE_SPELLS.includes(damageEvent?.ability.guid ?? -1)) {
+      this.handleHolyHeal(event);
+    }
+
+    if (SHADOW_DAMAGE_SPELLS.includes(damageEvent?.ability.guid ?? -1)) {
+      this.handleShadowHeal(event);
+    }
+  }
+
+  handleHolyHeal(event: AtonementAnalyzerEvent) {
+    const { healEvent } = event;
+    if (this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_HOLY_BUFF.id)) {
+      this.healing += calculateEffectiveHealing(healEvent, TE_INCREASE);
+    }
+  }
+
+  handleShadowHeal(event: AtonementAnalyzerEvent) {
+    const { healEvent } = event;
+    if (this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_SHADOW_BUFF.id)) {
+      this.healing += calculateEffectiveHealing(healEvent, TE_INCREASE);
+    }
+  }
+
+  onDamage(event: DamageEvent) {
+    if (
+      HOLY_DAMAGE_SPELLS.includes(event?.ability.guid ?? -1) &&
+      this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_HOLY_BUFF.id)
+    ) {
+      this.damage += calculateEffectiveDamage(event, TE_INCREASE);
+    }
+
+    if (
+      SHADOW_DAMAGE_SPELLS.includes(event?.ability.guid ?? -1) &&
+      this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_SHADOW_BUFF.id)
+    ) {
+      this.damage += calculateEffectiveDamage(event, TE_INCREASE);
+    }
+  }
+
+  onPTWApply(event: CastEvent | RefreshDebuffEvent | ApplyDebuffEvent) {
+    const targetString = encodeEventTargetString(event) || '';
+    if (this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_HOLY_BUFF.id)) {
+      this.ptwTargets.add(targetString);
+    } else if (event.type === EventType.ApplyDebuff) {
```

Comment:
@kfinch amended :)

Reply:
Since this behavior is so unusual, can you also leave a comment explaining why? Otherwise I think this PR is good to go